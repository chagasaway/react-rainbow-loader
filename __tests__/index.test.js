import React from 'react';
import RainbowLoader from '../src/index';
import renderer from 'react-test-renderer';

describe('RainbowLoader Component', () => {
  it('should not render', () => {
    const tree = renderer.create(
      <RainbowLoader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render', () => {
    const tree = renderer.create(
      <RainbowLoader visible={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with defaults', () => {
    const tree = renderer.create(
      <RainbowLoader visible={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const colors = ['#FFFFFF', '#000000'];
    const tree = renderer.create(
      <RainbowLoader
        visible={true} height='100%' duration={20} colors={colors}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
