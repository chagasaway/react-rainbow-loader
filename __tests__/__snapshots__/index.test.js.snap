// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RainbowLoader Component should not render 1`] = `null`;

exports[`RainbowLoader Component should not render 2`] = `null`;

exports[`RainbowLoader Component should render with custom props 1`] = `
<div
  className="RainbowLoader"
  style={
    Object {
      "animation": "rainbow 20s ease infinite",
      "backgroundImage": "linear-gradient(125deg, #FFFFFF, #000000)",
      "height": "100%",
    }
  }
/>
`;

exports[`RainbowLoader Component should render with defaults 1`] = `
<div
  className="RainbowLoader"
  style={
    Object {
      "animation": "rainbow 10s ease infinite",
      "backgroundImage": "linear-gradient(125deg, #FF0000, #FF7F00, #FFFF00, #00FF00, #0000FF, #4B0082, #8B00FF)",
      "height": "5px",
    }
  }
/>
`;
